src/eureka/S1_detector_processing/group_level.py

```diff
@@ -0,0 +1,181 @@
+#! /usr/bin/env python
+
+# File with functions to perform GLBS
+# trace masking, 
+# custom ref pixel correction for PRISM
+# EMMay, Nov 2022
+import numpy as np
+
+import scipy.signal as sgn
+import scipy.ndimage as spn
+
+from ..S3_data_reduction import background as bkg
+import astraeus.xarrayIO as xrio
+
+
+def GLBS(input_model, log, meta):
+    '''Function that performs group level.
+    background subtraction. Calls the S3 background code.
+
+    Parameters:
+    -----------
+    input_model : jwst.datamodels.QuadModel
+        The input group-level data product before ramp fitting.
+    log : logedit.Logedit
+        The open log in which notes from this step can be added.
+    meta : eureka.lib.readECF.MetaClass
+        The metadata object.
+
+    Returns:
+    --------
+    input_model : jwst.datamodels.QuadModel
+        The input group-level data product with background removed.
+    '''
+
+    log.writelog('  Running Group Level Background Subtraction.')
+
+    all_data = input_model.data
+    dq = input_model.groupdq
+
+    meta.inst = input_model.meta.instrument.name.lower()
+    meta.int_start = 0
+    meta.n_int = all_data.shape[0]
+
+    for ngrp in range(all_data.shape[1]):
+        grp_data = all_data[:, ngrp, :, :]
+        grp_mask = np.ones(grp_data.shape, dtype=bool)
+        dqmask = np.where((dq[:, ngrp, :, :] % 2 == 1) |
+                          (dq[:, ngrp, :, :] == 2))
+        grp_mask[dqmask] = 0
+
+        if hasattr(meta, 'masktrace') and meta.masktrace:
+            trace_mask = np.broadcast_to(input_model.trace_mask,
+                                         (grp_data.shape[0],
+                                          grp_data.shape[1],
+                                          grp_data.shape[2]))
+            trace_mask = np.nan_to_num(trace_mask).astype(bool)
+            grp_mask *= trace_mask
+
+        xrdata = (['time', 'y', 'x'], grp_data)
+        xrmask = (['time', 'y', 'x'], grp_mask)                
+        xrdict = dict(flux=xrdata, mask=xrmask)
+        data = xrio.makeDataset(dictionary=xrdict)
+        data['flux'].attrs['flux_units'] = 'n/a'
+        if ngrp == all_data.shape[1]-1:
+            data = bkg.BGsubtraction(data, meta, 
+                                     log, 
+                                     meta.isplots)
```

Comment:
The 4th input parameter is `m` (The current file/batch number). `isplots` isn't currently an input parameter.  See my previous comment from 14 days ago.